tests/integration/test_load_and_serve_agent.py

```diff
@@ -0,0 +1,185 @@
+)
+@pytest.mark.asyncio
+async def test_load_and_serve_agent(
+    agent_framework: AgentFramework, tmp_path: Path
+) -> None:
+    kwargs = {}
+
+    tmp_file = "tmp.txt"
+    logger.info("Starting")
+
+    if not check_uvx_installed():
+        msg = "uvx is not installed. Please install it to run this test."
+        raise RuntimeError(msg)
+
+    def write_file(text: str) -> None:
+        """write the text to a file in the tmp_path directory
+
+        Args:
+            text (str): The text to write to the file.
+
+        Returns:
+            None
+        """
+        with open(os.path.join(tmp_path, tmp_file), "w", encoding="utf-8") as f:
+            f.write(text)
+
+    kwargs["model_id"] = "gpt-4.1-mini"
+    env_check = validate_environment(kwargs["model_id"])
+    if not env_check["keys_in_environment"]:
+        pytest.skip(f"{env_check['missing_keys']} needed for {agent_framework}")
+
+    model_args: dict[str, Any] = (
+        {"parallel_tool_calls": False}
+        if agent_framework not in [AgentFramework.AGNO, AgentFramework.LLAMA_INDEX]
+        else {}
+    )
+    model_args["temperature"] = 0.0
+    tools = [
+        write_file,
+        MCPStdio(
+            command="uvx",
+            args=["mcp-server-time", "--local-timezone=America/New_York"],
+            tools=[
+                "get_current_time",
+            ],
+        ),
+    ]
+    test_tracer = InMemorySpanExporter()
+    additional_exporters=[test_tracer]
+    tracing_config = TracingConfig(additional_exporters=additional_exporters)
+    agent_config = AgentConfig(
+        tools=tools,  # type: ignore[arg-type]
+        instructions="Search the web to answer",
+        model_args=model_args,
+        **kwargs,  # type: ignore[arg-type]
+    )
+    agent_server = await AnyAgent.create_async(
+        agent_framework,
+        agent_config,
+        tracing=tracing_config,
+    )
+    try:
+        logger.info(f"Agent created: {agent_server}")  # noqa: G004
+        async_server = await agent_server.serve(
+            ServingConfig(port=5555, endpoint="/test_agent")
+        )
+        logger.info("Agent serving")
+
+        # TODO use an agent card instead
+
+        # open another call in another thread
+        client = A2AClient(url="http://localhost:5555/test_agent")
+        result = await client.send_task(
+            {
+                "id": "1",
+                "message": Message(
+                    role="user",
+                    parts=[
+                        TextPart(
+                            text="Use the tools to find what year it is in the America/New_York timezone and write the value (single number) to a file"
+                        )
+                    ],
+                ),
+            }
+        )
+        logger.info(f"after sending: {result}")  # noqa: G004
+        logger.info(f"after sending (traces):")
+        for s in test_tracer.get_finished_spans():  # noqa: G004
+            logger.info(s.to_json())
```

Comment:
Ahh! Ok now I understand: you added the ability to add exporters so that you can grab spans out of the test_tracer. However, would it not work to call `pop_trace` on the agent exporter that is already attached? It's not clear to me why a design change is necessary in order to access the trace.

Reply:
IMHO, because the exporter that _we_ add has its own purpose, and the user serving an agent may have the need to export traces for _their_ own purposes. Trace sampling, batching, and other practices could affect the way we run our business logic, but may be necessary in a specific deployment. In a similar way, we have our own exporter for our own purposes, which is not publicly documented in the agent, and does not use an existing api but instead our own api. Using the standard exporter api to add other exportes is IMHO the cleanest solution in this case.